src/ui/public/vis/vis_types/vislib_vis_legend.js

```diff
@@ -106,6 +106,9 @@ uiModules.get('kibana')
         };
 
         $scope.canFilter = function (legendData) {
+          if (['heatmap', 'gauge'].includes($scope.vis.vislibVis.visConfigArgs.type)) {
```

Comment:
Is this supposed to disable `heatmap` filtering as well? Because after testing this PR, I'm still able to filter on heatmaps.

Reply:
disables filttering on legend